src/Harmony-One/Signer.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Signer.h"
+#include "../Ethereum/RLP.h"
+#include "../HexCoding.h"
+
+using namespace TW;
+using namespace TW::Harmony;
+
+std::tuple<uint256_t, uint256_t, uint256_t> Signer::values(const uint256_t &chainID,
+                                                           const Data &signature) noexcept {
+    boost::multiprecision::uint256_t r, s, v;
+    import_bits(r, signature.begin(), signature.begin() + 32);
+    import_bits(s, signature.begin() + 32, signature.begin() + 64);
+    import_bits(v, signature.begin() + 64, signature.begin() + 65);
+    v += 27;
+
+    boost::multiprecision::uint256_t newV;
+    import_bits(newV, signature.begin() + 64, signature.begin() + 65);
+    newV += 35 + chainID + chainID;
+    return std::make_tuple(r, s, newV);
+}
+
+std::tuple<uint256_t, uint256_t, uint256_t>
+Signer::sign(const uint256_t &chainID, const PrivateKey &privateKey, const Data &hash) noexcept {
+    auto signature = privateKey.sign(hash, TWCurveSECP256k1);
+    return values(chainID, signature);
+}
+
+Proto::SigningOutput Signer::sign(const TW::Harmony::Proto::SigningInput &input) noexcept {
+    auto key = PrivateKey(Data(input.private_key().begin(), input.private_key().end()));
+    auto transaction = Transaction(
+        /* nonce: */ load(input.nonce()),
+        /* gasPrice: */ load(input.gas_price()),
+        /* gasLimit: */ load(input.gas_limit()),
+        /* fromShardID */ load(input.from_shard_id()),
+        /* toShardID */ load(input.to_shard_id()),
+        /* to: */ Address(input.to_address()),
+        /* amount: */ load(input.amount()),
+        /* payload: */ Data(input.payload().begin(), input.payload().end()));
+
+    auto signer = Signer(uint256_t(load(input.chain_id())));
+    signer.sign(key, transaction);
+    auto protoOutput = Proto::SigningOutput();
+    auto encoded = signer.rlp_no_hash(transaction, true);
+
+    protoOutput.set_encoded(encoded.data(), encoded.size());
+    auto v = store(transaction.v);
+    protoOutput.set_v(v.data(), v.size());
```

Comment:
let's just a signature field, we may also refactor Ethereum signing output

Reply:
so it's easier to handle

Reply:
I am not clear if I am should be making code changes to another project's src.

Reply:
why change other project src? it seems that we don't use r,s,v inside wallet core, so we can simplify it

Reply:
I thought the mentioning of "...we may also refactor Ethereum signing output" might imply me adding a commit for that work.